src/Microsoft.ML.Tokenizers/Model/CodeGen.cs

```diff
@@ -0,0 +1,1723 @@
+                IReadOnlyDictionary<string, int>? addedTokens = null,
+                bool addPrefixSpace = false,
+                bool addBeginningOfSentence = false,
+                bool addEndOfSentence = false,
+                string? unknownToken = DefaultSpecialToken,
+                string? beginningOfSentenceToken = DefaultSpecialToken,
+                string? endOfSentenceToken = DefaultSpecialToken) :
+            this(vocabularyPath is null ? throw new ArgumentNullException(nameof(vocabularyPath)) : File.OpenRead(vocabularyPath),
+                mergePath is null ? throw new ArgumentNullException(nameof(mergePath)) : File.OpenRead(mergePath),
+                preTokenizer, normalizer, addedTokens, addPrefixSpace, addBeginningOfSentence, addEndOfSentence, unknownToken, beginningOfSentenceToken, endOfSentenceToken, disposeStream: true)
+        {
+        }
+
+        /// <summary>
+        /// Construct tokenizer's model object to use with the English Robert model.
+        /// </summary>
+        /// <param name="vocabularyStream">The stream of a JSON file containing the dictionary of string keys and their ids.</param>
+        /// <param name="mergeStream">The stream of a file containing the tokens's pairs list.</param>
+        /// <param name="preTokenizer">The pre-tokenizer to use.</param>
+        /// <param name="normalizer">The normalizer to use.</param>
+        /// <param name="addedTokens">The additional tokens to add to the vocabulary.</param>
+        /// <param name="addPrefixSpace">Indicate whether to include a leading space before encoding the text.</param>
+        /// <param name="addBeginningOfSentence">Indicate whether to include the beginning of sentence token in the encoding.</param>
+        /// <param name="addEndOfSentence">Indicate whether to include the end of sentence token in the encoding.</param>
+        /// <param name="unknownToken">The unknown token.</param>
+        /// <param name="beginningOfSentenceToken">The beginning of sentence token.</param>
+        /// <param name="endOfSentenceToken">The end of sentence token.</param>
+        public CodeGen(
+                Stream vocabularyStream,
+                Stream mergeStream,
+                PreTokenizer? preTokenizer = null,
+                Normalizer? normalizer = null,
+                IReadOnlyDictionary<string, int>? addedTokens = null,
+                bool addPrefixSpace = false,
+                bool addBeginningOfSentence = false,
+                bool addEndOfSentence = false,
+                string? unknownToken = DefaultSpecialToken,
+                string? beginningOfSentenceToken = DefaultSpecialToken,
+                string? endOfSentenceToken = DefaultSpecialToken) :
+            this(vocabularyStream, mergeStream, preTokenizer, normalizer, addedTokens, addPrefixSpace, addBeginningOfSentence, addEndOfSentence, unknownToken, beginningOfSentenceToken, endOfSentenceToken, disposeStream: false)
```

Comment:
I'm approaching this similar to File APIs. The tokenizer simply reads the stream without any further manipulation. I find it cleaner not to transfer ownership of the stream, especially since it's straightforward for the user to dispose of it after creating the tokenizer. They can even employ a using statement on the stream for simplicity.